api/src/services/validation-service.ts

```diff
@@ -434,13 +436,13 @@ export class ValidationService extends DBService {
           )
         );
       });
-
-      if (!mediaState.isValid || csvState?.some((item) => !item.isValid)) {
-        // At least 1 error exists, skip remaining steps
-        parseError = true;
-      }
     });
 
+    if (!mediaState.isValid || csvState?.some((item) => !item.isValid)) {
+      // At least 1 error exists, skip remaining steps
+      parseError = true;
+    }
```

Comment:
We may as well just check this in the if below, and throw the SubmissionError directly, rather than assigning `parseError` here and then immediately checking it below.